compiler/src/dotty/tools/dotc/reporting/TestReporter.scala

```diff
@@ -0,0 +1,12 @@
+package dotty.tools
+package dotc
+package reporting
+
+import collection.mutable
+import Diagnostic._
+
+/** A re-usable Reporter used in Contexts#test */
+class TestingReporter extends StoreReporter(null):
+  infos = new mutable.ListBuffer[Diagnostic]
+  override def hasUnreportedErrors: Boolean = infos.exists(_.isInstanceOf[Error])
+  def reset(): Unit = infos.clear()
```

Comment:
A call to `removeBufferedMessages` will set `infos = null`, so this looks dangerous. I wonder if instead of this logic we could just have:
```scala
class TestingReporter extends StoreReporter(NoReporter)
```

Reply:
The idea was not to re-generate a ListBuffer for TestingReporter. I'll fix the problem by overriding `hasUnreportedErrors`.